components/encoder/client_test.go

```diff
@@ -94,21 +102,21 @@ func TestClient(t *testing.T) {
 
 	conn, err := viamgrpc.Dial(context.Background(), listener1.Addr().String(), logger)
 	test.That(t, err, test.ShouldBeNil)
-	workingEncoderClient := encoder.NewClientFromConn(context.Background(), conn, testEncoderName, logger)
+	workingEncoderClient := encoder.NewClientFromConn(context.Background(), conn, encoder.Named(testEncoderName), logger)
 
 	t.Run("client tests for working encoder", func(t *testing.T) {
 		// DoCommand
-		resp, err := workingEncoderClient.DoCommand(context.Background(), generic.TestCommand)
+		resp, err := workingEncoderClient.DoCommand(context.Background(), testutils.TestCommand)
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, resp["command"], test.ShouldEqual, generic.TestCommand["command"])
-		test.That(t, resp["data"], test.ShouldEqual, generic.TestCommand["data"])
+		test.That(t, resp["command"], test.ShouldEqual, testutils.TestCommand["command"])
+		test.That(t, resp["data"], test.ShouldEqual, testutils.TestCommand["data"])
 
 		err = workingEncoderClient.ResetPosition(context.Background(), nil)
 		test.That(t, err, test.ShouldBeNil)
 
 		pos, positionType, err := workingEncoderClient.GetPosition(
 			context.Background(),
-			nil,
+			encoder.PositionTypeUnspecified,
```

Comment:
do these changes make the `positionType` parameter required? can a user still pass in nothing or do they have to use `positionTypeUnspecified`?

Reply:
okay makes sense. I just ask because we discussed the optional functionality a lot, so I was curious about how this changed that. thanks!

Reply:
np!